google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/MessageReceiver.java

```diff
@@ -34,11 +34,10 @@
      */
     NACK
   }
-  
+
   /**
-   * Called when a message is received by the subscriber.
-   *
-   * @return A future that signals when a message has been processed.
+   * Called when a message is received by the subscriber. The implementation must arrange for {@code
+   * reponse} to be set after processing the {@code message}.
    */
-  ListenableFuture<AckReply> receiveMessage(PubsubMessage message);
+  void receiveMessage(PubsubMessage message, SettableFuture<AckReply> response);
```

Comment:
From what I understand, the 'final' doesn't change the signature - an interface can omit the `final`, and the implementations can add the `final`. 

http://stackoverflow.com/questions/5380177/final-arguments-in-interface-methods-whats-the-point